services/provider/server/server.go

```diff
@@ -278,23 +278,29 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		return nil, fmt.Errorf("secret %s data fsid is empty", monSecret)
 	}
 
-	// Service for monitoring endpoints
-	scMonitoring := &v1.Service{}
-	err = s.client.Get(ctx, types.NamespacedName{Name: "rook-ceph-mgr", Namespace: s.namespace}, scMonitoring)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get rook-ceph-mgr service. %v", err)
+	// Get mgr pod hostIP
+	podList := &corev1.PodList{}
+	err = s.client.List(ctx, podList, client.MatchingLabels(map[string]string{"app": "rook-ceph-mgr"}))
+	if err != nil || len(podList.Items) == 0 {
+		return nil, fmt.Errorf("failed to list pod with rook-ceph-mgr label. %v", err)
 	}
 
-	if scMonitoring.Spec.ClusterIP == "" || strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)) == "" {
-		return nil, fmt.Errorf("service rook-ceph-mgr clusterIP or port is empty")
+	var port int32
+	for i := range podList.Items[0].Spec.Containers[0].Ports {
+		if podList.Items[0].Spec.Containers[0].Ports[i].Name == "http-metrics" {
+			port = podList.Items[0].Spec.Containers[0].Ports[i].ContainerPort
+		}
```

Comment:
Let's make this a little be more readable 
```suggestion
		portSpec := &podList.Items[0].Spec.Containers[0].Ports;
		if portSpec.Name == "http-metrics" {
			port = portSpec.ContainerPort
		}
```